ValveResourceFormat/Resource/ResourceTypes/ModelAnimation/SegmentDecoders/CCompressedStaticVector3.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Numerics;
+using System.Linq;
+
+namespace ValveResourceFormat.ResourceTypes.ModelAnimation.SegmentDecoders
+{
+    public class CCompressedStaticVector3 : AnimationSegmentDecoder
+    {
+        public Vector3[] Data { get; }
+
+        public CCompressedStaticVector3(ArraySegment<byte> data, int[] elements, AnimationDataChannel localChannel) : base(elements, localChannel)
+        {
+            // Static data has only one frame of data, so prefetch all data to avoid unnecessary GC
+            Data = Enumerable.Range(0, data.Count / (3 * 2))
+                .Select(i =>
+                {
+                    var offset = i * (3 * 2);
+                    return new Vector3(
+                        (float)BitConverter.ToHalf(data.Slice(offset + (0 * 2))),
```

Comment:
Did you test that BitConverter.ToHalf produces same results as HalfTypeHelper.Convert?

Reply:
I tested it like that
```cs
DeltaData = Enumerable.Range(0, deltaData.Count / 2)
    .Select(i =>
    {
        var a = (float)BitConverter.ToHalf(deltaData.Slice(i * 2));
        var b = HalfTypeHelper.Convert(BitConverter.ToUInt16(deltaData.Slice(i * 2)));
        if (a != b)
            Debugger.Break();
        return a;
    })
    .ToArray();
```
and `Debugger.Break()` was indeed not called.
`HalfTypeHelper.Convert` implementation seems just about right for IEEE 754 half-precision float and C# `Half` is exactly that, so I guess HalfTypeHelper could be entirely removed, as VRF already targets .NET 6.

Reply:
Yeah, see #287.